app/services/dicom.js

```diff
@@ -4,21 +4,78 @@ const angular = require('angular');
 const app = angular.module('app');
 
 const fs = require('fs');
+const path = require('path');
+const zlib = require('zlib');
 const Rx = require('rx');
 const {dirListObs} = require('../common/util.js');
 const dicomParser = require('dicom-parser');
+const nifti = require('nifti-js');
 const TAG_DICT = require('../common/dataDictionary.js').TAG_DICT;
+const crypto = require('crypto');
+const filetypes = require('../common/filetypes.json');
+
+const extToScitranType = {};
+for (const type of Object.keys(filetypes)) {
+  for (const ext of filetypes[type]) {
+    extToScitranType[ext] = type;
+  }
+}
+
+const decompressForExt = {
+  '.gz': function gunzip(buffer) {
+    return zlib.gunzipSync(buffer);
+  }
+};
 
 function dicom($rootScope, organizerStore, fileSystemQueues) {
-  const parse = (filePath) => {
+  const parseHeadersForExt = {
+    '.nii': function(buffer) {
+      // TODO write a function that converts this into proper headers.
+      return nifti.parseNIfTIHeader(buffer);
+    },
+    '.dcm': function(buffer) {
```

Comment:
per IRL convo, I'll likely rip out this extension map. I've ended up implementing things differently in my engage branch.